haystack/nodes/label_generator/pseudo_label_generator.py

```diff
@@ -98,9 +107,9 @@ def __init__(
                 )
         else:
             raise ValueError("Provide either a QuestionGenerator or a non-empty list of questions/document pairs.")
-
+        self.devices, _ = initialize_device_settings(devices=devices, use_cuda=use_gpu, multi_gpu=False)
```

Comment:
Consider adding assert for `devices` to be of length 1 since we do not support multi GPU loading.

Reply:
+1, not assert but actual check as assert can be removed during code optimization.